### DISCUSSION TITLE
How to get EM loss results of icepak and maxwell coupling simulation using pyaedt

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#5278] [Q&A] [answered]
When I was using ansys electronics desktop for coupled simulation of icepak and maxwell, pyaedt scripts helped me a lot in automating the simulation process.
  But I recently came across a small requirement while post-processing. As shown in the figure below, two coupling iterations are performed, and my requirement is to get the EM loss value of the second iteration, such as “22.667 Watt” in the figure.

![image](https://github.com/user-attachments/assets/6b8915dc-f1cd-40f0-9538-a808fef08d23)

![image](https://github.com/user-attachments/assets/605cc85f-6b7e-42ac-a873-ce89f651c224)

![image](https://github.com/user-attachments/assets/210630e6-cf03-4eb0-a34f-ee5e46523946)

  The version of pyaedt I'm using is 0.9.10.
  I would be happy to know what code is needed to get the EM loss results of the coupled simulation of icepak and maxwell. Looking forward to your reply, thanks!

**Comment by lorenzovecchietti** [comment]
Did you try this method: https://aedt.docs.pyansys.com/version/stable/API/_autosummary/ansys.aedt.core.icepak.Icepak.assign_2way_coupling.html#ansys.aedt.core.icepak.Icepak.assign_2way_coupling ?

**Reply by aknsj** [reply]
Hello, thank you for your reply.
I have successfully setup the coupling calculation before,using the following code. 

```
ipk.assign_2way_coupling(number_of_iterations=2,ipk_iterations_per_coupling=100)
```


But my current requirement is to get the EM loss value of the last iteration.
Looking forward to your reply.

**Reply by lorenzovecchietti** [reply, marked as answer]
You should be able to retrieve the data from the field summary, like this: https://examples.aedt.docs.pyansys.com/version/dev/examples/electrothermal/graphic_card.html#Postprocess
modifying the add_calculation call:
https://aedt.docs.pyansys.com/version/stable/API/visualization/_autosummary/ansys.aedt.core.visualization.post.field_summary.FieldSummary.add_calculation.html

Something like:
```
fs = ipk.post.create_field_summary()
fs.add_calculation(entity="Boundary", geometry="Surface", geometry_name="...", quantity="HeatFlowRate")
```

Let me know if this works.

**Reply by aknsj** [reply]
Hi, I successfully got the EM loss using the following code. Thank you very much.
```
heatflowrate_fs = ipk.post.create_field_summary()
heatflowrate_fs.add_calculation(
        entity="Boundary",
        geometry="Surface",
        geometry_name=boundary_list,
        quantity="HeatFlowRate",
        side="Adjacent",
        mesh="All",
```
